src/components/KFileUpload/KFileUpload.vue

```diff
@@ -0,0 +1,344 @@
+<template>
+  <div
+    class="k-file-upload w-100"
+  >
+    <KLabel
+      v-if="label && appearance === 'file'"
+      :for="customInputId"
+      v-bind="labelAttributes"
+      data-testid="k-file-upload-label"
+    >
+      {{ label }}
+    </KLabel>
+
+    <KInput
+      :id="customInputId"
+      ref="fileInput"
```

Comment:
You're setting `ref="fileInput"` here; however, down in the `setup` function you actually set this `ref` to an array by default, meaning you're not using this template ref. Should this be removed? You can't do both.